crates/event-streaming/proto/walrus/event/v1/service.proto

```diff
@@ -0,0 +1,175 @@
+syntax = "proto3";
+
+package walrus.event.v1;
+
+import "google/api/annotations.proto";
+import "google/protobuf/empty.proto";
+import "google/protobuf/timestamp.proto";
+import "sui/rpc/v2beta/checkpoint_contents.proto";
+import "sui/rpc/v2beta/checkpoint_summary.proto";
+import "sui/rpc/v2beta/event.proto";
+
+// EventStreamingService provides a verifiable stream of events from the Sui
+// blockchain for an event stream. It allows clients to receive events
+// along with cryptographic proofs that tie them to a specific checkpoint.
+// It is designed to be forward-compatible, allowing for proof mechanisms
+// to be upgraded over time without breaking clients.
+service EventStreamingService {
+    // StreamEvents opens a server-side stream that sends sets of events
+    // as they are finalized in checkpoints. Each message in the stream contains
+    // all relevant events for a single checkpoint, plus the necessary data to verify them.
+    rpc StreamEvents(StreamEventsRequest) returns (stream VerifiableEvents) {
+        option (google.api.http) = {
+            post: "/v1/events:stream"
+            body: "*"
+        };
+    }
+
+    // GetNonInclusionProof allows a client to request a cryptographic proof
+    // that a specific event did NOT occur within a given range of checkpoints.
+    // This is a standard unary RPC call.
+    rpc GetNonInclusionProof(GetNonInclusionProofRequest) returns (GetNonInclusionProofResponse) {
+        option (google.api.http) = {
+            post: "/v1/events:getNonInclusionProof"
+            body: "*"
+        };
+    }
+}
+
+
+// ============== Event Streaming RPC Messages ==============
+
+// An enum defining the available proof types.
+enum ProofType {
+    // Defaults to the server's preferred/most modern proof type.
+    PROOF_TYPE_UNSPECIFIED = 0;
+
+    // The client does not require a proof.
+    PROOF_TYPE_NONE = 1;
+
+    // The V1 checkpoint contents proof.
+    PROOF_TYPE_CHECKPOINT_CONTENTS_V1 = 2;
+
+    // The V2 Merkle inclusion proof.
+    PROOF_TYPE_MERKLE_PROOF_V2 = 3;
+}
+
+message StreamEventsRequest {
+    // The checkpoint sequence number from which to start streaming.
+    // Events from checkpoints before this number will not be sent.
+    // Defaults to 0, starting from the first available checkpoint.
+    uint64 starting_checkpoint_sequence_number = 1;
```

Comment:
We're going to need to be very careful here when it comes to retention and how far back you can start the stream.

Reply:
Also everything that can needs to be prefixed with optional

Reply:
done (although it seems everything without optional is optional implicitly but i added it)